### ISSUE TITLE
mlx.core.isposinf has wrong output when input is mlx.core.int32

### ISSUE THREAD

**Issue by Redmept1on** [ml-explore/mlx#493]
**Describe the bug**
When it comes to the boundary value of mx.int32, 'mlx.core.isposinf' outputs a result different from numpy, while mx.int16 does not have this issue.

**To Reproduce**

Include code snippet
```python
import mlx.core as mx
import numpy as np

#int32
x = mx.array([-2147483648,0,2147483647], dtype=mx.int32)
result = mx.isposinf(x)
print("mlx:", result)

x = np.array([-2147483648,0,2147483647], dtype=np.int32)
print("numpy:", np.isposinf(x))

#int16
x = mx.array([-32768,0,32767], dtype=mx.int16)
result = mx.isposinf(x)
print("mlx:", result)

x = np.array([-32768,0,32767], dtype=np.int16)
print("numpy:", np.isposinf(x))
```
![image](https://github.com/ml-explore/mlx/assets/84373897/c02efce0-c052-4460-8b47-3bc671aa3ada)

**Expected behavior**
mlx: array([False, False, False], dtype=bool)

**Desktop (please complete the following information):**
 - OS Version: MacOS 14.2.1
 - Version 0.10.0

**Comment by awni** [comment]
That's a bug, thanks for finding and reporting! Fix here #494

**Comment by Redmept1on** [comment]
thx!
